optuna_dashboard/_app.py

```diff
@@ -139,8 +139,14 @@ def rename_study(study_id: int) -> dict[str, Any]:
             response.status = 404  # Not found
             return {"reason": f"study_id={study_id} is not found"}
 
+        summary = get_study_summary(storage, study_id)
+        if summary is None:
+            response.status = 500
+            return {"reason": "Failed to load the study"}
         try:
-            dst_study = optuna.create_study(storage=storage, study_name=dst_study_name)
+            dst_study = optuna.create_study(
+                storage=storage, study_name=dst_study_name, directions=summary.directions
+            )
```

Comment:
You can get `directions` of the source study from a `src_study` object.

```suggestion
            dst_study = optuna.create_study(
                storage=storage, study_name=dst_study_name, directions=src_study.directions
            )
```